### PR TITLE
chore(aws-cdk-readme): replace deprecated method used in aws-chatbot README.md

### PR DESCRIPTION
**Pull request by BLasan**
Currently addLambdaInvokeCommandPermissions method used to get the permissions,
which is a deprecated method now.
Use addToPolicy method to get necessary permissions

fix: #13444

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by BLasan** [comment]
@skinny85 Would you mind taking a look?

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 1]
```diff
@@ -34,7 +34,13 @@ const slackChannel = new chatbot.SlackChannelConfiguration(this, 'MySlackChannel
...
+const invokeCommandStatement = new iam.PolicyStatement({
+  effect: iam.Effect.ALLOW,
+  resources: ["RESOURCE_ARNS_ALLOWED"],
+  actions: ["INVOKING_ACTION_ALLOWED"],
+});
+
+slackChannel.addToRolePolicy(invokeCommandStatement);
 slackChannel.addNotificationPermissions();
 slackChannel.addSupportCommandPermissions();
 slackChannel.addReadOnlyCommandPermissions();
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:46]: We can remove these lines as the are also deprecated.

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 2]
```diff
@@ -34,7 +34,13 @@ const slackChannel = new chatbot.SlackChannelConfiguration(this, 'MySlackChannel
   slackChannelId: 'YOUR_SLACK_CHANNEL_ID',
 });
 
-slackChannel.addLambdaInvokeCommandPermissions();
+const invokeCommandStatement = new iam.PolicyStatement({
+  effect: iam.Effect.ALLOW,
+  resources: ["RESOURCE_ARNS_ALLOWED"],
+  actions: ["INVOKING_ACTION_ALLOWED"],
+});
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:41]: I think we can document this clearer and encourage a more idiomatic usage with the `grantX` methods on corresponding resources. IE, if we change this to.
```ts
const lambdaFn = new lambda.Function(this, 'MyChatbotHandler', {
  code: lambda.Code.fromAsset('./code'),
  runtime: lambda.Runtime.NODE_JS_14_X,
});
lambdaFn.grantInvoke(slackChannel.role);
```
@BLasan do you agree?

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 3]
```diff
@@ -34,7 +34,13 @@ const slackChannel = new chatbot.SlackChannelConfiguration(this, 'MySlackChannel
   slackChannelId: 'YOUR_SLACK_CHANNEL_ID',
 });
 
-slackChannel.addLambdaInvokeCommandPermissions();
+const invokeCommandStatement = new iam.PolicyStatement({
+  effect: iam.Effect.ALLOW,
+  resources: ["RESOURCE_ARNS_ALLOWED"],
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:39]: NitPick: use single quotes to match the general repo style. (I prefer double but what can ya do 🤷🏻‍♂️ )

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 2 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md:41]: Yuh, I think this would be ideal. But we need to add the proper permissions to the role first right?

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 3 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md:39]: My bad. Will change this :)

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 2 above)
Comment (MrArnoldPalmer) [reply packages/@aws-cdk/aws-chatbot/README.md:41]: so `lambdaFn.grantInvoke(slackChannel.role);` should handle all of that. This should be the only grant statement needed for the chatbot to invoke the function.

**Comment by BLasan** [comment]
@MrArnoldPalmer Updated

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 4]
```diff
@@ -27,17 +27,21 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
...
-slackChannel.addNotificationPermissions();
-slackChannel.addSupportCommandPermissions();
-slackChannel.addReadOnlyCommandPermissions();
+const invokeCommandStatement = new iam.PolicyStatement({
+  effect: iam.Effect.ALLOW,
+  resources: ['RESOURCE_ARNS_ALLOWED'],
+  actions: ['INVOKING_ACTION_ALLOWED'],
+});
+
+slackChannel.addToRolePolicy(invokeCommandStatement);
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:44]: the lambda grant invoke removes the need for this.

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 2 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md]: agreed

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 4 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md]: @MrArnoldPalmer Updated. please review

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 5]
```diff
@@ -27,25 +27,28 @@ This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aw
...
   ],
   resources: ['arn:aws:s3:::abc/xyz/123.txt'],
 }));
+
+const lambdaFn = new lambda.Function(this, 'MyChatbotHandler', {
+  code: lambda.Code.fromAsset('./code'),
+  runtime: lambda.Runtime.NODE_JS_14_X,
+  handler: 'FILE_NAME.handler',
+});
+lambdaFn.grantInvoke(slackChannel.role)
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:51]: Actually I'm not sure if this is needed here. I misinterpreted the point of the previous calls I believe. I'm not sure if `addLambdaInvokeCommandPermissions` is actually meant to grant access to a single specific lambda function. These aren't managed policies but just templates that the chatbot console will create a role with.

Perhaps we should just remove the usage of the methods that don't exist for now.

**Review by MrArnoldPalmer — DISMISSED** [review]

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 5 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md:51]: Yuh, that method is for granting permissions in order to access a particular lambda function. Should we remove this?

Comment (MrArnoldPalmer) [reply packages/@aws-cdk/aws-chatbot/README.md:51]: yeah lets remove this and the only changes from the original will be removing the calls to the deprecated methods. I do think users should be granting access to specific lambdas, cloudwatch metrics, etc and not using `*` resources. This however should be demonstrated in a more specific example.

**Comment by BLasan** [comment]
@MrArnoldPalmer Looks good?

File: packages/@aws-cdk/aws-chatbot/README.md [hunk 6]
```diff
@@ -34,10 +34,13 @@ const slackChannel = new chatbot.SlackChannelConfiguration(this, 'MySlackChannel
...
-slackChannel.addNotificationPermissions();
-slackChannel.addSupportCommandPermissions();
-slackChannel.addReadOnlyCommandPermissions();
+const invokeCommandStatement = new iam.PolicyStatement({
+  effect: iam.Effect.ALLOW,
+  resources: ['RESOURCE_ARNS_ALLOWED'],
+  actions: ['INVOKING_ACTION_ALLOWED'],
+});
+
+slackChannel.addToRolePolicy(invokeCommandStatement);
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-chatbot/README.md:43]: I think we should remove this as well. If you want to include examples about adding template policies, it should be done in a separate snippet with corresponding documentation in a different PR.

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-chatbot/README.md (hunk 6 above)
Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md:43]: Ok, will remove this as well :)

Comment (BLasan) [reply packages/@aws-cdk/aws-chatbot/README.md]: @MrArnoldPalmer Made the requested changes. Please review

**Review by MrArnoldPalmer — APPROVED** [review]

**Review by rix0rrr — APPROVED** [review]
